csrc/C++20/compare

```diff
@@ -0,0 +1,40 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+namespace std {
```

Comment:
Could you guard this with this macro?
https://github.com/NVIDIA/Fuser/blob/2abae36f572023a9c31200144997bd3e43a6b585/lib/dynamic_type/test/member.cpp#L76-L82

Reply:
Done. Thanks. I also cleaned up the simplifications you mentioned earlier.